--- a/ghc/lib/posix/Makefile
+++ b/ghc/lib/posix/Makefile
@@ -50,7 +50,7 @@ PosixErr_HC_OPTS      ='-\#include"cbits/libposix.h"'
 PosixFiles_HC_OPTS    ='-\#include"cbits/libposix.h"'
 PosixIO_HC_OPTS       ='-\#include"cbits/libposix.h"'
 PosixProcEnv_HC_OPTS  ='-\#include"cbits/libposix.h"'
-PosixProcPrim_HC_OPTS ='-\#include"cbits/libposix.h"'
+PosixProcPrim_HC_OPTS ='-\#include"cbits/libposix.h"' -monly-2-regs
 PosixTTY_HC_OPTS      ='-\#include"cbits/libposix.h"' -monly-2-regs
 Posix_HC_OPTS         ='-\#include"cbits/libposix.h"'
 
--- a/ghc/lib/std/Makefile
+++ b/ghc/lib/std/Makefile
@@ -53,7 +53,7 @@ Time_HC_OPTS        += -monly-3-regs -H16m
 
 # Far too much heap is needed to compile PrelNum with -O at the
 # moment, but there you go..
-PrelNum_HC_OPTS     += -H30m
+PrelNum_HC_OPTS     += -Onot -H30m
 
 PrelBase_HC_OPTS         += -H12m
 PrelRead_HC_OPTS         += -H13m
