--- a/pkgs/development/libraries/isl/default.nix
+++ b/pkgs/development/libraries/isl/default.nix
@@ -3,13 +3,7 @@
 stdenv.mkDerivation rec {
   name = "isl-0.07"; # CLooG 0.16.3 fails to build with ISL 0.08.
 
-  src = fetchurl {
-    urls = [
-        "http://www.kotnet.org/~skimo/isl/${name}.tar.bz2"
-        "ftp://ftp.linux.student.kuleuven.be/pub/people/skimo/isl/${name}.tar.bz2"
-      ];
-    sha256 = "0kpxmvhrwwdygqqafqzjf9xiksq7paac2x24g9jhr3f9ajj3zkyx";
-  };
+  src = ./isl-0.07.tar.bz2;
 
   buildInputs = [ gmp ];
 
