--- a/configure
+++ b/configure
@@ -6108,7 +6108,7 @@ if test "${libc_cv_broken_visibility_attribute+set}" = set; then :
 else
   cat > conftest.c <<EOF
 		  int foo (int x);
-		  int bar (int x) __asm__ ("foo") __attribute__ ((visibility ("hidden")));
+		  int bar (int x) __asm__ ("foo") __attribute__ ((visibility ("default")));
 		  int bar (int x) { return x; }
 EOF
 		  libc_cv_broken_visibility_attribute=yes
--- a/configure.in
+++ b/configure.in
@@ -1395,7 +1395,7 @@ EOF
     AC_CACHE_CHECK(whether __attribute__((visibility())) is supported,
 		 libc_cv_visibility_attribute,
 		 [cat > conftest.c <<EOF
-		  int foo __attribute__ ((visibility ("hidden"))) = 1;
+		  int foo __attribute__ ((visibility ("default"))) = 1;
 		  int bar __attribute__ ((visibility ("protected"))) = 1;
 EOF
 		  libc_cv_visibility_attribute=no
@@ -1418,7 +1418,7 @@ EOF
 		 libc_cv_broken_visibility_attribute,
 		 [cat > conftest.c <<EOF
 		  int foo (int x);
-		  int bar (int x) __asm__ ("foo") __attribute__ ((visibility ("hidden")));
+		  int bar (int x) __asm__ ("foo") __attribute__ ((visibility ("default")));
 		  int bar (int x) { return x; }
 EOF
 		  libc_cv_broken_visibility_attribute=yes
--- a/elf/rtld.c
+++ b/elf/rtld.c
@@ -142,7 +142,7 @@ struct rtld_global _rtld_global =
    declaration.  So spell out was strong_alias does plus add the
    visibility attribute.  */
 extern struct rtld_global _rtld_local
-    __attribute__ ((alias ("_rtld_global"), visibility ("hidden")));
+    __attribute__ ((alias ("_rtld_global"), visibility ("default")));
 
 
 /* This variable is similar to _rtld_local, but all values are
@@ -181,7 +181,7 @@ struct rtld_global_ro _rtld_global_ro attribute_relro =
    declaration.  So spell out was strong_alias does plus add the
    visibility attribute.  */
 extern struct rtld_global_ro _rtld_local_ro
-    __attribute__ ((alias ("_rtld_global_ro"), visibility ("hidden")));
+    __attribute__ ((alias ("_rtld_global_ro"), visibility ("default")));
 
 
 static void dl_main (const ElfW(Phdr) *phdr, ElfW(Word) phnum,
--- a/elf/tst-tls10.c
+++ b/elf/tst-tls10.c
@@ -1,7 +1,7 @@
 #include "tst-tls10.h"
 
 #ifdef USE_TLS__THREAD
-__thread int dummy __attribute__((visibility ("hidden"))) = 12;
+__thread int dummy __attribute__((visibility ("default"))) = 12;
 __thread struct A local = { 1, 2, 3 };
 #endif
 
--- a/elf/tst-tlsmod7.c
+++ b/elf/tst-tlsmod7.c
@@ -1,7 +1,7 @@
 #include "tst-tls10.h"
 
 #ifdef USE_TLS__THREAD
-__thread int dummy __attribute__((visibility ("hidden"))) = 12;
+__thread int dummy __attribute__((visibility ("default"))) = 12;
 __thread struct A a1 = { 4, 5, 6 };
 __thread struct A a2 = { 7, 8, 9 };
 __thread struct A a3 __attribute__((tls_model("initial-exec")))
--- a/elf/tst-tlsmod8.c
+++ b/elf/tst-tlsmod8.c
@@ -1,7 +1,7 @@
 #include "tst-tls10.h"
 
 #ifdef USE_TLS__THREAD
-__thread long long dummy __attribute__((visibility ("hidden"))) = 12;
+__thread long long dummy __attribute__((visibility ("default"))) = 12;
 __thread struct A a2 = { 22, 23, 24 };
 __thread struct A a4 __attribute__((tls_model("initial-exec")))
   = { 25, 26, 27 };
--- a/elf/tst-tlsmod9.c
+++ b/elf/tst-tlsmod9.c
@@ -1,7 +1,7 @@
 #include "tst-tls10.h"
 
 #ifdef USE_TLS__THREAD
-__thread int dummy __attribute__((visibility ("hidden"))) = 12;
+__thread int dummy __attribute__((visibility ("default"))) = 12;
 __thread struct A a1 = { 4, 5, 6 };
 __thread struct A a3 __attribute__((tls_model("initial-exec")))
   = { 10, 11, 12 };
--- a/include/libc-symbols.h
+++ b/include/libc-symbols.h
@@ -476,7 +476,7 @@ for linking")
   _INTVARDEF (name, name##_internal)
 # define _INTVARDEF(name, aliasname) \
   extern __typeof (name) aliasname __attribute__ ((alias (#name), \
-						   visibility ("hidden")));
+						   visibility ("default")));
 # define INTDEF2(name, newname) strong_alias (name, newname##_internal)
 # define INTVARDEF2(name, newname) _INTVARDEF (name, newname##_internal)
 #else
@@ -564,7 +564,7 @@ for linking")
 #if defined SHARED && defined DO_VERSIONING && !defined NO_HIDDEN
 # ifndef __ASSEMBLER__
 #  define __hidden_proto_hiddenattr(attrs...) \
-  __attribute__ ((visibility ("hidden"), ##attrs))
+  __attribute__ ((visibility ("default"), ##attrs))
 #  define hidden_proto(name, attrs...) \
   __hidden_proto (name, __GI_##name, ##attrs)
 #  define __hidden_proto(name, internal, attrs...) \
--- a/libio/stdio.c
+++ b/libio/stdio.c
@@ -43,7 +43,7 @@ _IO_FILE *stderr = (FILE *) &_IO_2_1_stderr_;
 # define AL(name) AL2 (name, _IO_##name)
 # define AL2(name, al) \
   extern __typeof (name) al __attribute__ ((alias (#name),                    \
-                                            visibility ("hidden")))
+                                            visibility ("default")))
 AL(stdin);
 AL(stdout);
 AL(stderr);
--- a/nptl/pthread_cond_timedwait.c
+++ b/nptl/pthread_cond_timedwait.c
@@ -38,7 +38,7 @@
 
 /* Cleanup handler, defined in pthread_cond_wait.c.  */
 extern void __condvar_cleanup (void *arg)
-     __attribute__ ((visibility ("hidden")));
+     __attribute__ ((visibility ("default")));
 
 struct _condvar_cleanup_buffer
 {
--- a/nptl/pthread_cond_wait.c
+++ b/nptl/pthread_cond_wait.c
@@ -37,7 +37,7 @@ struct _condvar_cleanup_buffer
 
 
 void
-__attribute__ ((visibility ("hidden")))
+__attribute__ ((visibility ("default")))
 __condvar_cleanup (void *arg)
 {
   struct _condvar_cleanup_buffer *cbuffer =
--- a/nptl/sysdeps/pthread/pt-initfini.c
+++ b/nptl/sysdeps/pthread/pt-initfini.c
@@ -66,7 +66,7 @@ static void
 call_initialize_minimal (void)
 {
   extern void __pthread_initialize_minimal_internal (void)
-    __attribute ((visibility ("hidden")));
+    __attribute ((visibility ("default")));
 
   __pthread_initialize_minimal_internal ();
 }
--- a/stdlib/longlong.h
+++ b/stdlib/longlong.h
@@ -904,7 +904,7 @@ UDItype __umulsidi3 (USItype, USItype);
 #define udiv_qrnnd(q, r, n1, n0, d) \
   do {									\
     extern UWtype __udiv_qrnnd_16 (UWtype, UWtype)			\
-                        __attribute__ ((visibility ("hidden")));	\
+                        __attribute__ ((visibility ("default")));	\
     /* r0: rn r1: qn */ /* r0: n1 r4: n0 r5: d r6: d1 */ /* r2: __m */	\
     __asm__ (								\
 	"mov%M4 %4,r5\n"						\
--- a/sysdeps/sparc/sparc32/bits/atomic.h
+++ b/sysdeps/sparc/sparc32/bits/atomic.h
@@ -60,7 +60,7 @@ __make_section_unallocated
 volatile unsigned char __sparc32_atomic_locks[64]
   __attribute__ ((nocommon, section (".gnu.linkonce.b.__sparc32_atomic_locks"
 				     __sec_comment),
-		  visibility ("hidden")));
+		  visibility ("default")));
 
 #define __sparc32_atomic_do_lock(addr) \
   do								      \
