--- a/pkgs/development/libraries/libssh2/default.nix
+++ b/pkgs/development/libraries/libssh2/default.nix
@@ -3,10 +3,7 @@
 stdenv.mkDerivation rec {
   name = "libssh2-1.2.6";
 
-  src = fetchurlBoot {
-    url = "${meta.homepage}/download/${name}.tar.gz";
-    sha256 = "00f6hw972v7jd0rrdr6kx5cn7pa1spyx8xgc7vhjydksgsig3f8b";
-  };
+  src = ./libssh2-1.2.6.tar.gz;
 
   buildInputs = [ openssl zlib ];
 
